.github/workflows/types.yml

```diff
@@ -0,0 +1,34 @@
+name: Types
+on: [push, pull_request]
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [lts/*]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install Gulp
+        run: npm install -g gulp-cli
+
+      - name: Install other dependencies
+        run: npm install
+
+      - name: Run typestest
```

Comment:
Similar to https://github.com/mozilla/pdf.js/pull/17263/files#diff-1704ff461d0e01b99f63a43be52b4fea0d33f7507044420d57d94fd7a52855aeR59 let's call this `Run types tests` (I'll use the same naming scheme of "Run <name> tests" for the other upcoming workflows).